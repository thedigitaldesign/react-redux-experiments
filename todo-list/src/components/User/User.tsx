import React, { Component } from 'react'

interface Props {
  store?: any
}

export class User extends Component<Props> {
  render() {
    return (
      <div {...this.props}>
        <div className="field">
          <label className="label">Username or Todo List name</label>
          <div className="control">
            <input type="text" className="input user-input" placeholder="Team_Voldemort" />
          </div>
        </div>

        <div className="field">
          <div className="control">
            <button className="button is-link is-pulled-right">
              Continue
            </button>
          </div>
        </div>
      </div>
    )
  }
}
