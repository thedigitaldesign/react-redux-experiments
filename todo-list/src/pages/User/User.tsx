import React, { Component } from 'react'

// Packages

// Components
import { User as UserComponent } from '../../components'

export default class User extends Component {
  render() {
    return (
      <>
        <UserComponent />
      </>
    )
  }
}
