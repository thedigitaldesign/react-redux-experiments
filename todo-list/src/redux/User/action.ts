// Types
import * as type from '../types'

// Models
import { UserModel } from '../../models'

export const GetUsername = () => {
  return {
    type: type.UsernameTypes.USERNAME
  }
}

export const SetUsername = (username: UserModel) => {
  return {
    type: type.UsernameTypes.USERNAME,
    payload: username
  }
}