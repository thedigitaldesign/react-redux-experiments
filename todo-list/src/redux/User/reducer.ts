// Models
import { UserModel } from '../../models'

// Types
import * as type from '../types'

export default (state: any, action: { type: string; payload: UserModel | string }) => {
  switch(action.type) {
    case type.UsernameTypes.USERNAME:
      return action.payload
    default:
      return state || null
  }
}
