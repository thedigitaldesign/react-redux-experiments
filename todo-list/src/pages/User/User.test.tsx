import React from 'react'

// Packages
import Enzyme, {shallow} from 'enzyme'

// Routes
import User from './User'

describe('Pages > User', () => {
  const user = shallow(<User />)

  it('should render pages User', () => {
    expect(user).toMatchSnapshot()
  });
});
