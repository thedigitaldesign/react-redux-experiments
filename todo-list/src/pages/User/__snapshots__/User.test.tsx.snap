// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pages > User should render pages User 1`] = `ShallowWrapper {}`;
